### ISSUE TITLE
3.3.7 Accessible Authentication queries on personal info

### ISSUE THREAD

**Issue by alastc** [w3c/wcag#1320]
From Aaron Armstrong:
<hr>
This feedback is re: 3.3.7 Accessible Authentication, which states "If an authentication process relies on a cognitive function test, at least one other method must also be available that does not rely on a cognitive function test." Cognitive function tests include "memorization, such as remembering a username, password..."

https://www.w3.org/TR/2020/WD-WCAG22-20200811/#dfn-cognitive-function-test

However, the Sufficient Techniques [permit](https://www.w3.org/WAI/WCAG21/Understanding/accessible-authentication) "providing a properly marked up email and password inputs."

A couple issues I perceived upon first reading:

1.	One criteria mentions "username" and the other mentions "email." Unless the intent is to specifically recommend the use of email addresses and not unique usernames for login purposes, this wording should perhaps be standardized.
2.	More importantly, the criteria seem to list user/pass fields as both sufficient login technique and also a cognitive function test. I believe the intent here is to emphasize "properly marked up," which would allow the use of tools like saving login credentials in the browser or using external password managers and bypass the user being required to remember them. If this is the case, I would suggest expanding upon this in the Sufficient Techniques section and making these expectations clear. 

I hope this is helpful. Thank you for the work you're doing!

Aaron

**Comment by alastc** [comment]
There is a pull request to update the definition to make the difference clearer between personal info and site specific info in [#1412](https://github.com/w3c/wcag/pull/1412/files).

I'll try and put something together for the email/username differences.

**Comment by alastc** [comment]
For the email/username aspect, we aren't trying to say anything specific about those, and the highlighted note in the understanding starts with:
> Web sites can employ username (or email) and password inputs as an authentication method...

The only difference between a username and email address (used as username) is that remembering your own email is not considered a cognitive function test. However, there is inevitably a password or other means of authenticating, so it doesn't make much difference in general.

I'm not seeing anything that needs updating, unless you can expand on what the misunderstanding is?

**Comment by wardav** [comment]
To you think remembering and inputting things such as:

> Things which are about you, and do not vary by site. E.g. Email address, Name, postal address.

Should be in the normative scope. Just looking at that tweak #1412, I don't know that there is some clear language explaining that they are in scope / not in scope. 

For example

> Remembering a password is a cognitive function test. Such tests are known to be problematic for many people with cognitive disabilities. Whether it is remembering random strings of characters, a pattern gesture to perform on a touch screen, or identifying which images include a particular object, cognitive function tests will exclude some people. When a cognitive function test is used, at least one other authentication method must be available which is not a cognitive function test.

> Remembering an email address is not a cognitive test. While users may user different emails between websites, the user should be aware of their own email address. The same is true with other personal information that does not reasonably change that may be included in the authentication process. For example, entering memeorable information, such as a favourite pet.

**Comment by alastc** [comment]
A change we made yesterday was to the [definition of cognitive function test](https://github.com/w3c/wcag/blob/master/guidelines/terms/22/cognitive-function-test.html):
> memorization, such as remembering a username, password, set of characters, images, or patterns. **The common identifiers name, e-mail, and phone number are not considered cognitive function tests as they are personal to the user and consistent across websites**;

Does that work for you?

**Comment by alastc** [comment]
Closing, I responded to the original poster via email.
